tephra-core/src/main/java/org/apache/tephra/coprocessor/TransactionStateCache.java

```diff
@@ -184,4 +185,14 @@ private void refreshState() throws IOException {
   public TransactionVisibilityState getLatestState() {
     return latestState;
   }
+
+  protected void setId(@Nullable String id) {
+    if (id != null) {
+      this.logPrefix = "[" + id + "] ";
+    }
+  }
+
+  private String prefixLog(String message) {
```

Comment:
not sure this is a very good idea. It means you are performing the string operations even when it is not being logged (for example, for debug messages). Better to add the logPrefix as an argument to the log message, such as:
```
LOG.debug("[{}] Latest transaction snapshot: {}", logPrefix, latestState.toString()));
```

Reply:
HBase co-processor uses Apache commons-logging. The `{}` notation is not available in the commons-logging APIs.

I couldn't figure out a way to not create a string unless absolutely necessary. The only debug log in that file has `LOG.isDebugEnabled()` guard. However, I know this is not an ideal solution.

Reply:
I was not aware of that. I think this is not logged very frequently, so it's ok.